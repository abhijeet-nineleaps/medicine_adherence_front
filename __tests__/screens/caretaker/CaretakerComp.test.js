import React from 'react';
import enableHooks from "jest-react-hooks-shallow";
import Caretakercomp from '../../../src/screens/caretaker/CaretakerComp';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({adapter: new Adapter()});

enableHooks(jest);
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn().mockImplementation((func) => func()),
}));
jest.mock("react-native-elements", () => ({
  default: jest.fn(),
  Tab: {
    Item : () => ({}),
  },
  TabView: {
    Item: () => ({}),
  }
}));
jest.mock("react-native-paper", () => ({
  default: jest.fn(),
}));
const findNodeByTestId = (wrapper, testID) => {
  return wrapper.findWhere((node) => {
    return node.prop("testID") === testID;
  });
};
const props = {
  check2: true,
  navigation: {
    addListener: jest.fn(),
    navigate: jest.fn(),
  },
};
describe('Click send image', () => {
  it('renders correctly', () => {
    
    const wrapper  = shallow(<Caretakercomp/>);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call onDeleteList", () => {
    const checkforlogin = true;

    React.useState = jest.fn().mockReturnValue([checkforlogin, {}]);
    const wrapper = shallow(<Caretakercomp props={props} checkforlogin={true} />);
    const tree = findNodeByTestId(wrapper, "loginF");
    tree.props().onPress();
  });
 
  

});