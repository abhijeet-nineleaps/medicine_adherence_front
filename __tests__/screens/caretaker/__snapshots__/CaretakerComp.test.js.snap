// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `ShallowWrapper {}`;
